File: test/e2e/redact.go
package e2e

func redactLoreArgs(args []string) []string {
	out := append([]string(nil), args...)
	for i := 0; i < len(out); i++ {
		if out[i] == "--token" && i+1 < len(out) {
			out[i+1] = "<redacted>"
			i++
		}
	}
	return out
}

File: crates/lore-auth-e2e/tests/redact.rs
mod support;

#[test]
fn redact_lore_args_hides_token_value() {
    if !support::require_e2e() {
        return;
    }

    let args = support::redact_lore_args(&[
        "auth",
        "login",
        "--token-type",
        "lore",
        "--token",
        "header.payload.signature",
        "--auth-url",
        "https://localhost:8081",
    ]);

    let joined = args.join(" ");
    assert!(
        !joined.contains("header.payload.signature"),
        "redacted args leaked token: {joined}"
    );
    assert!(
        joined.contains("<redacted>"),
        "redacted args missing marker: {joined}"
    );
}
